### DISCUSSION TITLE
[Win 10/11] Almost done building, but getting 2 errors with Resource Compiler and non-existing header

### DISCUSSION THREAD

**Discussion by Nevdogg** [AcademySoftwareFoundation/OpenRV#313] [Build Issues]
Hey all,

So I have been spending the last couple days trying to get this to build and I feel I am very close, hoping to get a pointer. I have been using info from this repo as well as richardssam's fork with his windows-config branch (which is what im on). I am on Windows 11. I have VS 2017 Professional (with Win 10 SDK 10.0.17763.0) and VS 2019 Community (with Win 10 SDK 10.0.19041.0 and Win 11 SDK 10.0.22000.0). I have both SDKs for 2019 b/c I was previously getting one error that 10.0.22000.0 was missing.

It looks like I just have 2 errors left but I can't seem to figure out how to fix these:

```
-- Build files have been written to: C:/OpenRV/_build/RV_DEPS_ZLIB/src
         Performing build step for 'RV_DEPS_ZLIB'
         [1/37] RC C:\OpenRV\_build\RV_DEPS_ZLIB\src\win32\zlib1.rc CMakeFiles\zlib.dir\win32\zlib1.res.d CMakeFiles\zlib.dir\win32\zlib1.res "Note: including file: " "C:/Program Files (x86)/Microsoft Visual Studio/2017/Professional/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe" C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x86\rc.exe -DNO_FSEEKO -DZLIB_DLL -D_CRT_NONSTDC_NO_DEPRECATE -D_CRT_SECURE_NO_DEPRECATE -I C:\OpenRV\_build\RV_DEPS_ZLIB\src -DWIN32 /fo CMakeFiles\zlib.dir\win32\zlib1.res C:\OpenRV\_build\RV_DEPS_ZLIB\src\win32\zlib1.rc
         FAILED: CMakeFiles/zlib.dir/win32/zlib1.res 
         RC C:\OpenRV\_build\RV_DEPS_ZLIB\src\win32\zlib1.rc CMakeFiles\zlib.dir\win32\zlib1.res.d CMakeFiles\zlib.dir\win32\zlib1.res "Note: including file: " "C:/Program Files (x86)/Microsoft Visual Studio/2017/Professional/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe" C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x86\rc.exe -DNO_FSEEKO -DZLIB_DLL -D_CRT_NONSTDC_NO_DEPRECATE -D_CRT_SECURE_NO_DEPRECATE -I C:\OpenRV\_build\RV_DEPS_ZLIB\src -DWIN32 /fo CMakeFiles\zlib.dir\win32\zlib1.res C:\OpenRV\_build\RV_DEPS_ZLIB\src\win32\zlib1.rc
         Microsoft (R) Windows (R) Resource Compiler Version 10.0.10011.16384
         Copyright (C) Microsoft Corporation.  All rights reserved.
     5>CUSTOMBUILD : fatal error RC1107: invalid usage; use RC /? for Help [C:\OpenRV\_build\cmake\dependencies\RV_DEPS_ZLIB.vcxproj]
```

       146>CustomBuild:
         1>Automatic MOC and UIC for target MuQt5

         AutoGen error
         -------------
         Info error in info file
         "C:/OpenRV/_build/src/lib/mu/MuQt5/CMakeFiles/MuQt5_autogen.dir/AutogenInfo.json":
         The header file "SRC:/src/lib/mu/MuQt5/MuQt5/generated/moc_SignalSpy_filtered.hpp" does not exist.
		 
         The system cannot find the batch label specified - VCEnd
	    146>C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(195,5): error MSB6006: "cmd.exe" exited with code 1. [C:\OpenRV\_build\src\lib\mu\MuQt5\MuQt5_autogen.vcxproj]
	    146>Done Building Project "C:\OpenRV\_build\src\lib\mu\MuQt5\MuQt5_autogen.vcxproj" (default targets) -- FAILED.

**Comment by OlivierArgentieri** [comment]
Hey ! 

I think i have the same errors, i have tried with VS2022, 2019 and 2017 and same errors for some library, here is the details:

(vs 2022: 6errors like this)
` fatal error RC1107: invalid usage; use RC /? for Help [C:\OpenRV\_build\cmake\dependencies\RV_DEPS_ZLIB.vcxproj` 
 
(vs 2019: 10errors and i got this)
`ninja : error : loading 'build.ninja': Le fichier spÚcifiÚ est introuvable. [C:\OpenRV\_build\cmake\dependencies\RV_DEPS_JPEGTURBO.vcxproj]`

(vs 2017: 5errors and i got this)
`ninja : error : loading 'build.ninja': Le fichier spÚcifiÚ est introuvable. [C:\OpenRV\_build\cmake\dependencies\RV_DEPS_JPEGTURBO.vcxproj]`


Maybe compilation works only on some specfici version of microsoft sdk or msbuild ? 
I have this on my different try:

VS2017:
```
Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.19045.
Build Engine version 15.9.21+g9802d43bc3 pour .NET Framework
```

VS2019:
```
Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.19045.
Build Engine version 16.11.2+f32259642 pour .NET Framework
```

VS2022:
```
Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.19045
Version MSBuild 17.7.2+d6990bcfa pour .NET Framework
```

I don't know if my error are related to yours, but we got the same message `CUSTOMBUILD : fatal error RC1107: invalid usage; use RC /? for Help [C:\OpenRV\_build\cmake\dependencies\RV_DEPS_ZLIB.vcxproj]`
